lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -257,12 +570,19 @@ def test_find(self, selinux_enabled, server_config, make_logger, tarball):
         assert list(new.datasets) == [md5]
         assert list(new.tarballs) == [dataset_name]
 
-    def test_lifecycle(self, selinux_enabled, server_config, make_logger, tarball):
+    def test_lifecycle(
+        self, selinux_enabled, server_config, make_logger, tarball, monkeypatch
+    ):
         """
         Create a dataset, unpack it, remove the unpacked version, and finally
         delete it.
         """
 
+        def mock_run(args, **kwargs):
```

Comment:
It would be good to have a comment here which explains that, while most of this test is actually functional, this mock prevents the tarball contents from actually being unpacked.

Ditto at line 498.